models/DataObject/ClassDefinition/Data/Time.php

```diff
@@ -113,16 +113,16 @@ public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
             throw new Model\Element\ValidationException('Wrong time format given must be a 5 digit string (eg: 06:49) [ '.$this->getName().' ]');
         }
 
-        if (!$omitMandatoryCheck && strlen($data)) {
+        if (!$omitMandatoryCheck && is_string($data) && strlen($data)) {
             if (!$this->toTime($data)) {
                 throw new Model\Element\ValidationException('Wrong time format given must be a 5 digit string (eg: 06:49) [ '.$this->getName().' ]');
             }
 
-            if (strlen($this->getMinValue()) && $this->isEarlier($this->getMinValue(), $data)) {
+            if (is_string($this->getMinValue()) && strlen($this->getMinValue()) && $this->isEarlier($this->getMinValue(), $data)) {
                 throw new Model\Element\ValidationException('Value in field [ '.$this->getName().' ] is not at least ' . $this->getMinValue());
             }
 
-            if (strlen($this->getMaxValue()) && $this->isLater($this->getMaxValue(), $data)) {
+            if (is_string($this->getMaxValue()) && strlen($this->getMaxValue()) && $this->isLater($this->getMaxValue(), $data)) {
```

Comment:
```suggestion
            if ($this->getMaxValue() && $this->isLater($this->getMaxValue(), $data)) {
```
Should be `string|null`, so I think it is enough to check it without `strlen()`